internal/clients/builder.go

```diff
@@ -38,120 +42,117 @@ and APIs available in Azure Stack via Azure Stack Profiles.
 	if err != nil {
-		return nil, fmt.Errorf("unable to get MSAL authorization token for batch management API: %+v", err)
+		return nil, fmt.Errorf("unable to find environment %q from endpoint %q: %+v", builder.AuthConfig.Environment.Name, builder.MetadataHost, err)
 	}
+	resourceManagerEndpoint, _ := builder.AuthConfig.Environment.ResourceManager.Endpoint()
 
-	keyVaultAuth = builder.AuthConfig.MSALBearerAuthorizerCallback(ctx, environment.KeyVault, sender, oauthConfig, string(environment.KeyVault.Endpoint))
+	account, err := NewResourceManagerAccount(ctx, resourceManagerAuth, *builder.AuthConfig, builder.SubscriptionID, builder.SkipProviderRegistration, *azureEnvironment, *resourceManagerEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("building account: %+v", err)
+	}
 
-	// Helper for obtaining endpoint-specific tokens
-	tokenFunc = func(endpoint string) (autorest.Authorizer, error) {
-		api := environments.Api{Endpoint: environments.ApiEndpoint(endpoint)}
-		authorizer, err := builder.AuthConfig.GetMSALToken(ctx, api, sender, oauthConfig, endpoint)
-		if err != nil {
-			return nil, fmt.Errorf("getting MSAL authorization token for endpoint %s: %+v", endpoint, err)
-		}
-		return authorizer, nil
+	client := Client{
+		Account: account,
 	}
 
 	o := &common.ClientOptions{
-		SubscriptionId:              builder.AuthConfig.SubscriptionID,
-		TenantID:                    builder.AuthConfig.TenantID,
-		PartnerId:                   builder.PartnerId,
-		TerraformVersion:            builder.TerraformVersion,
-		KeyVaultAuthorizer:          keyVaultAuth,
-		ResourceManagerAuthorizer:   auth,
-		ResourceManagerEndpoint:     env.ResourceManagerEndpoint,
-		StorageAuthorizer:           storageAuth,
-		SynapseAuthorizer:           synapseAuth,
-		BatchManagementAuthorizer:   batchManagementAuth,
-		SkipProviderReg:             builder.SkipProviderRegistration,
-		DisableCorrelationRequestID: builder.DisableCorrelationRequestID,
+		Environment: builder.AuthConfig.Environment,
+		Features:    builder.Features,
+
+		SubscriptionId:   builder.SubscriptionID,
+		TenantId:         builder.AuthConfig.TenantID,
+		PartnerId:        builder.PartnerID,
+		TerraformVersion: builder.TerraformVersion,
+
+		BatchManagementAuthorizer: batchManagementAuth,
+		KeyVaultAuthorizer:        keyVaultAuth,
+		ResourceManagerAuthorizer: resourceManagerAuth,
+		StorageAuthorizer:         storageAuth,
+		SynapseAuthorizer:         synapseAuth,
+
+		BatchManagementAutorestAuthorizer: authWrapper.AutorestAuthorizer(batchManagementAuth),
+		KeyVaultAutorestAuthorizer:        authWrapper.AutorestAuthorizer(keyVaultAuth).BearerAuthorizerCallback(),
+		ResourceManagerAutorestAuthorizer: authWrapper.AutorestAuthorizer(resourceManagerAuth),
+		StorageAutorestAuthorizer:         authWrapper.AutorestAuthorizer(storageAuth),
+		SynapseAutorestAuthorizer:         authWrapper.AutorestAuthorizer(synapseAuth),
```

Comment:
renaming these is going to involve both changing the files in this repository and the generated logic in `hashicorp/pandora` - instead I think we'd be better to introduce the new authorizers in a struct instead? e.g.

```suggestion

		Authorizers Authorizers
	}
	// ...
	
	type Authorizers struct {
		BatchManagementDataPlane SomeType
		KeyVaultDataPlane        SomeType
		ResourceManager SomeType,
		StorageDataPlane SomeType
		SynapseDataPlane SomeType
	}
```

which means we could leave the existing ones as they are and remove the old ones over time - which also gives us the option to refresh all the tokens at once if required?